packages/dynamite/dynamite/lib/src/models/type_result/some_of.dart

```diff
@@ -0,0 +1,128 @@
+part of 'type_result.dart';
+
+@immutable
+abstract class TypeResultSomeOf extends TypeResult {
+  TypeResultSomeOf(
+    super.className, {
+    required this.subTypes,
+    super.nullable,
+  }) : super(isTypeDef: false);
+
+  final Set<TypeResult> subTypes;
+
+  @override
+  String? get _builderFactory => null;
+
+  @override
+  String? get _serializer {
+    if (optimizedSubTypes.length <= 1) {
+      return null;
+    }
+
+    return '..add(${className}Extension.serializer)';
+  }
+
+  late final BuiltSet<TypeResult> optimizedSubTypes = _optimizedSubTypes.toBuiltSet();
+
+  @override
+  TypeResult get dartType {
+    if (optimizedSubTypes.isEmpty) {
+      throw StateError('SomeOf must have at least one sub type.');
+    }
+
+    if (optimizedSubTypes.length == 1) {
+      return optimizedSubTypes.single;
+    }
+
+    final record = optimizedSubTypes.map((final type) {
+      final dartType = type.nullableName;
+      final dartName = toDartName(dartType);
+      final fieldName = toFieldName(dartName, type.className);
+
+      return '$dartType $fieldName';
+    });
+
+    return TypeResultBase('({${record.join(',')}})');
+  }
+
+  Iterable<TypeResult> get _optimizedSubTypes sync* {
+    var optimizedNum = false;
+
+    final numbers = subTypes.where(
```

Comment:
```suggestion
    final numbersCount = subTypes.where(
```

Reply:
this is a collection thus I'd not call it `count`

Reply:
Ah true, but since you only ever access the length you could only store that instead

Reply:
It looks ugly like that 😅

Reply:
Ok 🤷‍♀️